src/app/reporting/Engine.h

```diff
@@ -40,10 +42,30 @@
 namespace chip {
 namespace app {
 
-class InteractionModelEngine;
 class TestReadInteraction;
 
 namespace reporting {
+
+// An object support for functionality required by reporting.
+class EngineDelegate
+{
+public:
+    virtual ~EngineDelegate()                                                                                     = default;
+    virtual Messaging::ExchangeManager * GetExchangeManagerForReportingEngine()                                   = 0;
```

Comment:
Why is this not just called GetExchangeManager()?  I understand that practically it's because we then have a name "collision", but why do we care about that in this API definition?  And in the implementation the two things are signature-compatible, no?